pkg/util/iterutil/iterutil_test.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package iterutil
+
+import (
+	"testing"
+
+	"github.com/cockroachdb/errors"
+)
+
+func TestIter(t *testing.T) {
+	input := []string{"a", "b", "c", "d", "e", "f", "g", "h"}
+
+	var output []string
+	fn := func(s *State) error {
+		str, ok := s.Cur().(string)
```

Comment:
suggestion: we use `require`-style testing in our modern code (where we don't, it's either old code or personal pref of the author). So you could write this as

```go
require.True(ok, "unexpected type %T", s.Cur())
```

and similar comments apply throughout the test (`require.NoError`, etc). But - this is not mandatory, just pointing it out.

Reply:
Oh okay! I must have seen some old code then. I'll update this.